### PR TITLE
Realize non-default port guest connect on broker

### PR DESCRIPTION
**Pull request by yanpliu**
Realize the non-default port 22 guest connect on broker.

Cases PASS for default port 22 connect and non-default guest connect by command.

### CODE REVIEW

File: broker/session.py [hunk 1]
```diff
@@ -28,7 +28,8 @@ def __init__(self, **kwargs):
         user = kwargs.get("username", "root")
         sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         sock.settimeout(kwargs.get("timeout"))
-        simple_retry(sock.connect, [(host, kwargs.get("port", 22))])
+        port = kwargs.get("port", "port") or 22
```
Comment (jyejare) [inline broker/session.py:31]:
```suggestion
        port = kwargs.get("port", 22)
```

File: broker/session.py [hunk 2]
```diff
@@ -28,7 +28,8 @@ def __init__(self, **kwargs):
...
+        simple_retry(sock.connect, [(host, kwargs.get("port", port))])
```
Comment (jyejare) [inline broker/session.py:32]:
```suggestion
        simple_retry(sock.connect, [(host, port)])
```

**Review by jyejare — CHANGES_REQUESTED** [review]
ACK pending comments !

File: broker/settings.py [hunk 3]
```diff
@@ -19,6 +19,7 @@
     Validator("HOST_USERNAME", default="root"),
     Validator("HOST_PASSWORD", must_exist=True),
     Validator("HOST_CONNECTION_TIMEOUT", default=None),
+    Validator("PORT", default=22),
```
Comment (JacobCallahan) [inline broker/settings.py:22]:
```suggestion
    Validator("HOST_SSH_PORT", default=22),
```

File: broker/hosts.py [hunk 4]
```diff
@@ -22,6 +22,7 @@ def __init__(self, hostname, name=None, from_dict=False, **kwargs):
         self.timeout = kwargs.get(
             "connection_timeout", settings.HOST_CONNECTION_TIMEOUT
         )
+        self.port = kwargs.get("port", settings.PORT)
```
Comment (JacobCallahan) [inline broker/hosts.py:25]:
```suggestion
        self.port = kwargs.get("port", settings.HOST_SSH_PORT)
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
Please use `HOST_SSH_PORT`as the settings name (clarity) and add it to the example settings file.
Thanks for contributing to Broker!

File: broker/session.py (hunk 1 above)
Comment (yanpliu) [reply broker/session.py]: > 
@jyejare 
Thanks for your suggestion, have already updated.

File: broker/session.py (hunk 2 above)
Comment (yanpliu) [reply broker/session.py]: > 
@jyejare 
Thanks for your suggestion, have already updated.

File: broker/settings.py (hunk 3 above)
Comment (yanpliu) [reply broker/settings.py]: > 
@JacobCallahan 
Thanks for your suggestion, have already updated.

File: broker/hosts.py (hunk 4 above)
Comment (yanpliu) [reply broker/hosts.py]: > 
@JacobCallahan 
Thanks for your suggestion, have already updated.

**Comment by yanpliu** [comment]
@JacobCallahan  @jyejare Thank you very much for your suggestion, I have updated the changes.
I have one question about the PR process. I am not sure whether I did the right steps or not.
When I did the update and submit the changes, I found the "Commit Suggestion" buttons are gray.
If I choose the "Commit suggestion" , the code can be updated and commit to the PR,  I do not need to update and git commit from my local dev env, Is that right?

**Comment by JacobCallahan** [comment]
@yanpliu I believe that is more of a repository permissions issue than anything you're doing wrong. Thanks for updating

**Comment by yanpliu** [comment]
> broker_settings.yaml.example
@JacobCallahan Thanks for your reply, what should I add to the broker_settings.yaml.example file? 
I  have go through the file  broker_settings.yaml.example ,
```
Container:
    instances:
        - docker:
            host_username: "<username>"
            host_password: "<plain text password>"
            host_port: None
            default: True
        - remote:
            host: "<remote hostname>"
            host_username: "<username>"
            host_password: "<plain text password>"
            **host_port: "<plain text port>"**----------------------------> add port in here?
```

**Comment by JacobCallahan** [comment]
@yanpliu try near the top under `HOST_PASSWORD`

**Comment by yanpliu** [comment]
> HOST_PASSWORD

Do you mean add 
```
# Host Settings
host_username: "root"
host_password: "<password>"
host_port:"<port>"  ---------> add here?
``

**Comment by yanpliu** [comment]
> host_port to broker_settings.yaml.example file

@JacobCallahan Have added host_port to broker_settings.yaml.example file

File: broker_settings.yaml.example [hunk 5]
```diff
@@ -4,6 +4,7 @@ inventory_file: "inventory.yaml"
 # Host Settings
 host_username: "root"
 host_password: "<password>"
+host_port: "<port>"
```
Comment (JacobCallahan) [inline broker_settings.yaml.example:7]: this should be host_ssh_port, correct?
```suggestion
host_ssh_port: "<port>"
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
one minor change

**Comment by yanpliu** [comment]
> one minor change

@JacobCallahan  Have already updated host_ssh_port in broker_settings.yaml.example file

**Review by JacobCallahan — APPROVED** [review]
ACK, thanks for the PR!

**Pull request merged**